File: apps/ai-service/app/schemas/field_map.py
from pydantic import BaseModel, Field
from typing import Optional
from .provider import ProviderConfigRequest


class FieldMapRequest(BaseModel):
    page_schema: dict = Field(default_factory=dict, description="Extracted page structure")
    form_schema: dict = Field(default_factory=dict, description="Extracted form fields")
    page_html: str = Field(default="", description="Raw page HTML for AI to parse")
    profile: dict = Field(..., description="User profile data")
    resume: dict = Field(..., description="Parsed resume data")
    provider: ProviderConfigRequest


class FieldMapping(BaseModel):
    field_id: str = ""
    selector: str = ""
    value: str = ""
    confidence: float = 0.0
    source: str = ""


class FieldMapResponse(BaseModel):
    mappings: list[FieldMapping] = []
    unknown_fields: list[str] = []
    confidence_overall: float = 0.0

File: apps/api/Application/DTOs/AI/FieldMap.cs
namespace BrowserAgent.Api.Application.DTOs.AI;

public class FieldMapRequest
{
    public Dictionary<string, object> PageSchema { get; set; } = new();
    public Dictionary<string, object> FormSchema { get; set; } = new();
    public Dictionary<string, object> Profile { get; set; } = new();
    public Dictionary<string, object> Resume { get; set; } = new();
    public ProviderConfigDto Provider { get; set; } = null!;
}

public class FieldMapResponse
{
    public List<FieldMapping> Mappings { get; set; } = new();
    public List<string> UnknownFields { get; set; } = new();
    public double ConfidenceOverall { get; set; }
}

public class FieldMapping
{
    public string FieldId { get; set; } = string.Empty;
    public string Value { get; set; } = string.Empty;
    public double Confidence { get; set; }
    public string Source { get; set; } = string.Empty;
}
